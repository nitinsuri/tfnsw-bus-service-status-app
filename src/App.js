import React, { useState } from 'react';
import PropTypes from 'prop-types';
import useFetch from './customHooks/useFetch';
import { appStaticLables } from './constants';
import Header from './Components/Header/Header';
import Footer from './Components/Footer/Footer';
import Alerts from './Components/Alert/Alert';
import Loader from './Components/Loader/Loader';
import AccordionPanel from './Components/AccordionPanel/AccordionPanel';
import './style.scss';

const App = () => {
  const { data, loading, error } = useFetch('./bus.json');
  const [expanded, setExpanded] = useState('panel0');
  const { sectionTitle } = appStaticLables;

  const handleAccordionChange = (panel) => (event, isExpanded) => {
    setExpanded(isExpanded ? panel : false);
  };
  return (
    <>
      <Header />
      <main>
        <h1>{sectionTitle}</h1>
        {error && <Alerts severity={'error'} message={error} />}
        {loading && <Loader />}
        {data?.map((service, idx) => (
          <>
            <AccordionPanel
              service={service}
              idx={idx}
              expanded={expanded}
              handleAccordionChange={handleAccordionChange}
            />
          </>
        ))}
      </main>
      <Footer />
    </>
  );
};
App.propTypes = {
  appStaticLables: PropTypes.object,
  error: PropTypes.string,
  loading: PropTypes.bool,
  data: PropTypes.object,
  handleAccordionChange: PropTypes.func,
};

export default App;
