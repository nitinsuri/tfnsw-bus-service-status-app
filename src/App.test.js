import { render } from '@testing-library/react'
import App from './App'

test('loads and displays the app', () => {
  const app = render(<App />)
  expect(app).toMatchSnapshot()
})
