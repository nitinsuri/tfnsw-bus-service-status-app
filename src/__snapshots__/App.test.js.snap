// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays the app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <nav>
          <h1>
            <img
              src="https://www.transport.nsw.gov.au/themes/tfnsw_corp_theme/source/tfnsw/components/header/images/logo-TfNSW.png"
            />
          </h1>
        </nav>
      </header>
      <main>
        <h1>
          Bus Report
        </h1>
        <div
          id="loader"
        >
          <span />
          <span />
          <span />
          <span />
          <span />
        </div>
      </main>
      <footer>
        <p>
          Copyright 
          2023
           Transport for NSW.
        </p>
      </footer>
    </div>
  </body>,
  "container": <div>
    <header>
      <nav>
        <h1>
          <img
            src="https://www.transport.nsw.gov.au/themes/tfnsw_corp_theme/source/tfnsw/components/header/images/logo-TfNSW.png"
          />
        </h1>
      </nav>
    </header>
    <main>
      <h1>
        Bus Report
      </h1>
      <div
        id="loader"
      >
        <span />
        <span />
        <span />
        <span />
        <span />
      </div>
    </main>
    <footer>
      <p>
        Copyright 
        2023
         Transport for NSW.
      </p>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
